src/App.tsx

```diff
@@ -17,18 +88,29 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterPosts={handleChangeFilter}
+                textFilter={handleChangeText}
+                isHaveText={textFilter}
+                clearButton={handleClearButton}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {filterLoading && <Loader />}
+              <TodoList
+                todos={loadedTodos}
+                handleClick={setIsLoadings}
+                filterPost={filterByPost}
+                selectedPost={postetTodoId}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {isLoading && (
```

Comment:
you can get rid of `isLoading` state and use `!!postetTodoId` here
That will allow you to simplify the logic, remove some functiuons